main/src/ca/uwaterloo/flix/tools/Summary.scala

```diff
@@ -346,7 +339,7 @@ object Summary {
       if (lines != other.lines) {
         val src = debugSrc.getOrElse(unknownSource)
         throw InternalCompilerException(s"lines '$lines' and '${other.lines}' in $debugSrc",
-          SourceLocation(isReal = true, SourcePosition.firstPosition(src), SourcePosition.firstPosition(src))
+          SourceLocation(isReal = true, src, SourcePosition.FirstPosition, SourcePosition.FirstPosition)
```

Comment:
Unrelated to your changes, shouldn't this function assert that the sources are equal too? The doc comment says that it combines file data from the *same file*.

Reply:
Hmm maybe, but this is just Summary and mostly exists for a paper artifact. Should probably be cut down a bit now that its published